src/Presentation/RestApi/Validator/Json/JsonSchemaValidator.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Presentation\RestApi\Validator\Json;
+
+use JsonSchema\Validator;
+use Symfony\Component\HttpFoundation\Response;
+
+class JsonSchemaValidator
+{
+    /**
+     * @var \JsonSchema\Validator
+     */
+    protected Validator $validator;
+
+    /**
+     * @param \JsonSchema\Validator $validator
+     */
+    public function __construct(Validator $validator)
+    {
+        $this->validator = $validator;
+    }
+
+    /**
+     * @param array $request
+     *
+     * @return array
+     */
+    public function validate(array $request): array
+    {
+        $request = json_decode((string)json_encode($request), false);
```

Comment:
Why do you use it?

Reply:
`\JsonSchema\Validator` requires `$request` as object type.

Reply:
pls change public function validate(Request $request): array
and move validation from the listener to this validator

Reply:
Fixed